include/pros/device.hpp

```diff
@@ -23,6 +23,28 @@
 
 namespace pros {
 inline namespace v5 {
+
+/*
+ * Enum of possible v5 devices.
+ *
+ * Contains all current V5 Devices.
+ */
+enum class DeviceType {
+	none = 0,
+	motor = 2,
+	rotation = 4,
+	imu = 6,
+	distance = 7,
+	radio = 8,
+	vision = 11,
+	adi = 12,
+	optical = 16,
+	gps = 20,
+	serial = 129,
+	generic __attribute__((deprecated("use E_DEVICE_SERIAL instead"))) = serial,
```

Comment:
this isn't necessary, the only reason this is "deprecated" is because it was old API. Since this is new API we do not need to start with a deprecated member. 

```suggestion
	generic __attribute__((deprecated("use E_DEVICE_SERIAL instead"))) = serial,
```